gerrit_python_tools/gerrit.py

```diff
@@ -624,13 +682,30 @@ def send_upstream(self, downstream, upstream):
         approvals = self.get_approvals(downstream.SSH())
 
         # Check to see if comment has necessary approvals.
-        if not self.is_upstream_approved(approvals):
-            msg = ("Could not send to upstream: One or more labels"
-                   " not approved.")
+        if self.is_forced():
+            if not self.is_release_approved(approvals):
+                msg = "Could not send to upstream: Release not approved."
+                logger.debug("Change %s: %s" % (self.change_id, msg))
+                ssh.exec_once('gerrit review -m %s %s'
+                              % (pipes.quote(msg), self.revision))
+                return
+            msg = ("***WARNING***\n%s (%s) has requested a forced upstream "
+                   "push. Bypassing all votes except for Release...\n\n"
+                   "Current votes are:\n\n%s"
+                   % (self._data['author']['name'],
+                      self._data['author']['email'],
+                      self.stringify_approvals(approvals)))
             logger.debug("Change %s: %s" % (self.change_id, msg))
             ssh.exec_once('gerrit review -m %s %s'
                           % (pipes.quote(msg), self.revision))
-            return
+        else:
+            if not self.is_upstream_approved(approvals):
```

Comment:
I think you might be confused on how this system works.  It's not providing any votes, it's simply reading them to ensure the correct votes are there before pushing the code upstream.  It has a method that mimics what gerrit does internally which is to check that the MAX value of a particular column, whether that is +1 or +2 or whatever, is present and that the MIN value (-1 or -2 or whatever) is not present.  This entire method is triggered on all "comment-added" events in gerrit.  It then verifies if the project is even an upstream project, then it checks to see if the push trigger is present (+2 Verify typically, or now the force comment), then it checks to see if the appropriate approvals are in place (+1 release for the force, +2 CR/QA/V and +1R for the typical path).  If those conditions are met, it uses the username of the committer of the latest patchset to push the code to upstream gerrit and then leaves a comment in gerrit to let you know what the upstream link is.